apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -146,18 +147,25 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
   }
 
   /**
-   * Serializes the PNPM Shrinkwrap file
+   * Gets the version number from the list of top-level dependencies in the "dependencies" section
+   * of the shrinkwrap file
    */
-  protected serialize(): string {
-    return yaml.safeDump(this._shrinkwrapJson, SHRINKWRAP_YAML_FORMAT);
+  public getTopLevelDependencyVersion(dependencyName: string): string | undefined {
+    return BaseShrinkwrapFile.tryGetValue(this._shrinkwrapJson.dependencies, dependencyName);
   }
 
   /**
-   * Gets the version number from the list of top-level dependencies in the "dependencies" section
-   * of the shrinkwrap file
+   * Gets the id of the specified package
    */
-  protected getTopLevelDependencyVersion(dependencyName: string): string | undefined {
-    return BaseShrinkwrapFile.tryGetValue(this._shrinkwrapJson.dependencies, dependencyName);
+  public getPackageId(packageName: string): string | undefined {
```

Comment:
What is the "ID?" Can you provide some better documentation and/or a better function name?

Reply:
Will do.

Reply:
I was using the id field in the shrinkwrap to get the path to a project tarball, but found that there is a field called `tarball` which is precisely what I need.